File: python/2019/day_01.py
from pathlib import Path


def main():
    text = (Path(__file__).parent / "../../input/2019/input_01.txt").read_text()
    module_fuel = 0
    total_fuel = 0
    for line in text.splitlines():
        mass = int(line)
        fuel = mass // 3 - 2
        module_fuel += fuel
        while fuel > 0:
            total_fuel += fuel
            fuel = fuel // 3 - 2
    print("Part 1:", module_fuel)
    print("Part 2:", total_fuel)


if __name__ == "__main__":
    main()

File: rust/2019/src/bin/day_01.rs
use std::fs;

fn main() {
    let text = fs::read_to_string("../input/2019/input_01.txt").unwrap();
    let module_fuel: i32 = text.lines().map(|line| line.parse::<i32>().unwrap() / 3 - 2).sum();
    println!("Part 1: {}", module_fuel);
    let total_fuel: i32 = text.lines().map(|line| {
        let mut total = 0;
        let mut mass = line.parse::<i32>().unwrap() / 3 - 2;
        while mass > 0 {
            total += mass;
            mass = mass / 3 - 2;
        }
        total
    }).sum();
    println!("Part 2: {}", total_fuel);
}
